_rules/table-header-cell-has-assigned-cells-d0f69e.md

```diff
@@ -27,20 +27,19 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any HTML element with the [semantic role][] of [rowheader][] or [columnheader][] for which all of the following is true:
+This rule applies to any HTML element with the [semantic role][] of [rowheader][] or [columnheader][] within a [table][] or [grid][] element, where the [table][] or [grid][] element is [visible][].
 
-- the element is [visible][]; and
-- the element is [included in the accessibility tree][]; and
-- the element has at least one ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][]; and
-- the element's closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][] is [included in the accessibility tree][].
+## Expectation 1
 
-## Expectation
+Each target element is [assigned][] to at least one element with a [semantic role][] of [cell][]. The test target and the [cell][] are within an element with the [semantic role][] of [table][] or [grid][].
 
-Each target element is [assigned][] to at least one element with a [semantic role][] of [cell][] or inheriting from [cell][].
+## Expectation 2
+
+Each target element is [assigned][] to at least one element with a [semantic role][] of [gridcell][]. The test target and the [gridcell][] are within an element with the [semantic role][] of [grid][].
 
 ## Assumptions
 
-This rule assumes that table header cells have a relationship conveyed through presentation with other cells within the same table. This excludes edge cases such as a table definition where there is only one header cell, or a table definition where there are multiple headers and no other cells. In such scenarios the rule fails, but [success criterion 1.3.1 Info and Relationships][sc1.3.1] could still be satisfied.
+This rule assumes that table header cells have a relationship conveyed through presentation with data cells within the same table. This excludes edge cases, where there are multiple headers and no data cells. In such scenarios the rule fails, but [success criterion 1.3.1 Info and Relationships][sc1.3.1] could still be satisfied.
```

Comment:
I agree with https://github.com/act-rules/act-rules.github.io/issues/1545#issuecomment-777604045 that we could move part of this Assumption to Applicability by requiring that the table/grid contains at least one non-empty cell/gridcell.

Reply:
What if "table header has assigned at least one element with the semantic role of `cell` that has non-empty accessible name"? 
There might be a non-empty data cell but if it has `aria-hidden` than that would also be a failure under 1.3.1.

Reply:
If we apply this rule to either a `rowheader` or a `columnheader` and either of those has assigned at least one non-empty element with a semantic role of `cell` or inheriting from cell than that should work. What do you think?

Reply:
What we need to avoid here is
```html
<table>
<tr><th>Foo</th></tr>
</table>
```
Where the `th` is not assigned to any `cell`, but this is OK because there is no relationship conveyed through presentation, so 1.3.1 doesn't care. But this does fail the rule because the `th` is assigned to no cell.
Rather than having it in an assumption (actually, in addition to the assumption which covers other corner cases), we could have that bit in the Applicability:
"the `table` contains at least one data cell".
This would remove from the start a large chunk of the corner cases covered by this assumption.